flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManagerDriver.java

```diff
@@ -75,7 +80,7 @@
 	/** Current max pod index. When creating a new pod, it should increase one. */
 	private long currentMaxPodId = 0;
 
-	private KubernetesWatch podsWatch;
+	private Optional<KubernetesWatch> podsWatch;
```

Comment:
The `podsWatch` related changes should be in a separate commit. They are hotfix for exist codes and is unrelated to the purpose of this PR.